### PR TITLE
fix: moving refreshTokenClientCredentialsLocation from oAuth2Common to AuthoCode

### LINKED ISSUE #12957: [Bug] : OAuth2- Authenticated API | Client Credentials
The datasource creation when selected grant type as Client Credentials fails.

Scenario 1 - The datasource creation happens but it suddenly starts failing with  and then you are no longer able to add the datasource.

`{"responseMeta":{"status":500,"success":false,"error":{"code":5000,"message":"Internal server error while processing request"}}}`

Scenario 2 - When the datasource is created and you would try to update it, the save and re-authorize will not work and throw below exception.

At the moment, due to these bugs a user will not be able to proceed and is blocked for client credentials flow.

### PR DESCRIPTION
**Pull request by rashmi-rai**
## Description

Moving `refreshTokenClientCredentialsLocation` from `oAuth2Common` to `AuthorizationCode` interface as it is required for `AuthorizationCode` only.

Fixes #12957 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: fix/client-credentials-flow 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | app/client/src/utils/WorkerUtil.ts | 89.76 **(0)** | 70.59 **(-1.96)** | 100 **(0)** | 93.33 **(0)**</details>

### CODE REVIEW

File: app/client/src/transformers/RestAPIDatasourceFormTransformer.ts [hunk 1]
```diff
@@ -213,6 +211,8 @@ const datasourceToFormAuthentication = (
           typeof authentication.isAuthorizationHeader === "undefined"
             ? true
             : !!authentication.isAuthorizationHeader,
+        refreshTokenClientCredentialsLocation:
+          authentication.refreshTokenClientCredentialsLocation || "",
```
Comment (nidhi-nair) [inline app/client/src/transformers/RestAPIDatasourceFormTransformer.ts:215]: Can we avoid the backup value of empty string and use `BODY` instead here?

**Review by nidhi-nair — APPROVED** [review]

**Comment by Aishwarya-U-R** [comment]
Failing cypress test 1. [Bug]- 11415 - Open Modal from button and test scroll running at Smoke_TestSuite/ClientSideTests/DisplayWidgets/Canvas_scrolling_spec.js is flaky. Giving a fix soon. 
Please proceed.

**Comment by nidhi-nair** [comment]
Merging in as Aishwarya has confirmed that the failing test is flaky.

**Pull request merged**